roles/openshift_version/tasks/main.yml

```diff
@@ -9,24 +9,43 @@
 # be used by default. Users must indicate what they want.
 - fail:
     msg: "Must specify openshift_release or openshift_image_tag in inventory to install origin. (suggestion: add openshift_release=\"1.2\" to inventory)"
-  when: is_containerized | bool and openshift.common.deployment_type == 'origin' and openshift_release is not defined and openshift_image_tag is not defined
+  when:
+  - is_containerized | bool
+  - openshift.common.deployment_type == 'origin'
+  - openshift_release is not defined
+  - openshift_image_tag is not defined
 
 # Normalize some values that we need in a certain format that might be confusing:
 - set_fact:
     openshift_release: "{{ openshift_release[1:] }}"
-  when: openshift_release is defined and openshift_release[0] == 'v'
+  when:
+  - openshift_release is defined
+  - openshift_release[0] == 'v'
 
 - set_fact:
     openshift_release: "{{ openshift_release | string }}"
   when: openshift_release is defined
 
+# Verify that the image tag is in a valid format
+- assert:
+    that:
+      - "{{ openshift_image_tag|match('(^[v]?\\d+\\.\\d+[\\.\\d+]*-?\\w*$)') }}"
```

Comment:
> Does that regex match e.g. v1.5.0-alpha.3? Is there a semver/other version parsing util we can use, to say "starts with v and the rest is a valid version" without regex?

It wouldn't. It would match v1.5.0-alpha3 though. I can easily add a ``.`` if that is part of the versioning scheme here.

AFAIK there is not another util we can use without pulling in yet another library into the installer which we are trying to avoid.

Good catch on latest. I'll add a ``when`` that will skip this if ``latest`` is used.